### ISSUE TITLE
After `!update` bot still does not know that crystal spray is bugged even though it is in modo-bugs

### ISSUE THREAD

**Issue by vorpal-buildbot** [PennyDreadfulMTG/Penny-Dreadful-Tools#4377]
Reported on Discord by bakert#2193

**Comment by bakert** [comment]
I cannot repro this on local. I see the Crystal Spray bug on my local copy of the bot. `!restartbot` also does not clear the issue.

The process is:

1. Person says "!update"
2. command.update executes
3. multiverse.update_bugged_cards is called
4. fetcher.bugged_cards is called
5. fetcher_internal.fetch_json is called
6. JSON is parsed and inserted into db
7. [Crystal Spray] is called and the bug is still not there.

So something in that chain is either caching or bugged.

https://pennydreadfulmtg.github.io/modo-bugs/bugs.json has the right info.

The database on prod has the bug against Crystal Spray's card_id:

    | 181557 |   13873 | Cards that ask you to choose "a basic land type or a color word" only let you choose a basic land type. | Disadvantageous |     1518670911 | https://github.com/PennyDreadfulMTG/modo-bugs/issues/295 |             0 |

**Comment by bakert** [comment]
Prod db.

```
MariaDB [cards]> SELECT bugs FROM _cache_card WHERE name = 'Crystal Spray';
+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| bugs                                                                                                                                                                                          |
+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Cards that ask you to choose "a basic land type or a color word" only let you choose a basic land type.|Disadvantageous|1518670911|https://github.com/PennyDreadfulMTG/modo-bugs/issues/295|0 |
+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
1 row in set (0.00 sec)
```

OK this is getting silly now.

**Comment by bakert** [comment]
Right now you can see this phenomenon in production with Hellion Eruption.

It has a bug that gives all the right details and has all the right labels - https://github.com/PennyDreadfulMTG/modo-bugs/issues/589

but `!update` doesn't shift it.

**Comment by bakert** [comment]
As before:

- https://pennydreadfulmtg.github.io/modo-bugs/bugs.json has the Hellion Eruption bug.
- Works correctly on local.
- prod cards db has Hellion Eruption entry in `card_bug` - `1| 411760 |   75532 | Hellion Eruption doesn't give you tokens and sacrifices everything you have                                                                                                                                                                                                                                                                                                                                                                                                                               | Disadvantageous         |     1535182672 | https://github.com/PennyDreadfulMTG/modo-bugs/issues/589 |             0 |        0 |`
- _cache_card has the bug entry - `| 75532 | normal |   75532 | Hellion Eruption | {5}{R}    |    6 | NULL  | NULL      | NULL    | Sorcery | Sacrifice all creatures you control, then create that many 4/4 red Hellion creature tokens. | Sacrifice all creatures you control, then create that many 4/4 red Hellion creature tokens. | hellion eruption | NULL | NULL | NULL    | 1        | Hellion Eruption | 75532   | Hellion Eruption | Commander:Legal,Legacy:Legal,Modern:Legal,Vintage:Legal,Penny Dreadful:Legal,Penny Dreadful EMN:Legal,Penny Dreadful KLD:Legal,Penny Dreadful AER:Legal,Penny Dreadful AKH:Legal,Penny Dreadful HOU:Legal,Penny Dreadful XLN:Legal,Penny Dreadful RIX:Legal,Penny Dreadful DOM:Legal |        1 | Hellion Eruption doesn't give you tokens and sacrifices everything you have |Disadvantageous|1535182672|https://github.com/PennyDreadfulMTG/modo-bugs/issues/589|0|0 |`

However:

- restartbot did fix it this time

**Comment by bakert** [comment]
Results for matching [Card Name] come (ultimately) from oracle.cards_by_name which uses the module-level var CARDS_BY_NAME which is populated only when it is entirely empty by oracle.init.

So I guess we have this global-ish var full of all the card details. We call `update` which calls

```
oracle.legal_cards(force=True)
multiverse.update_bugged_cards()
multiverse.update_cache()  
```

none of which depopulates oracle.CARDS_BY_NAME.

Essentially `update` causes an update of the database but not of what we have loaded into our module-level var. Added a force param in oracle.init and called it directly at the end of `update` in 6602eed1.
